src/cljfx/mutator.clj

```diff
@@ -58,8 +58,21 @@
                       ^ListChangeListener %2)))
 
 (defn observable-list [get-list-fn]
-  (let [set-all! #(.setAll ^ObservableList (get-list-fn %1)
-                           ^Collection %2)]
+  (let [set-all! (fn [instance coll]
+                   (let [l ^ObservableList (get-list-fn instance)
+                         size (.size l)
+                         csize (count coll)]
+                     ;; update the first (count coll) elements in ObservableList
+                     (reduce (fn [_ [idx e]]
+                               (if (< idx size)
+                                 (when (not (identical? e (.get l idx)))
+                                   (.set l idx e))
+                                 (.add l idx e)))
+                             nil
+                             (map-indexed vector coll))
```

Comment:
I'd prefer to not create an intermediate collection, transducers maybe? Or rewrite as `loop` for extra efficiency?